rfcs/054-id-resolver/README.md

```diff
@@ -0,0 +1,155 @@
+
+### Type Inference
+
+Currently, transformers infer the type of a Concept from its use in the input. This can lead to incorrect and
+inconsistent types being applied.
+
+In MARC, for example, `x00` fields contain "Personal Name" fields, so are expected to contain Concepts of type Person.
+Places can be found in `651` (Geographic Term) fields, etc.  In these cases, the Authoritative Identifier for the Concept 
+is extracted from the `$0` subfield.
+
+However, this is neither validated before reaching the pipeline, nor are these rules reliable.
+
+In practice, a `700` field may contain a Person, or it may contain a text attributed to the named person.  Which one it is
+can only be known by checking the identifier from `$0` against a known record. (see the Malmonides Problem, below)
+
+### LoC namespace
+
+MARC data from Sierra does not distinguish between LCSH and LCNames. They are both signalled by the use of `indicator2=0`.
+As with type, above, the namespace is inferred from the field.  In "Name" fields (`x00`, `x10`, `x11`), it is assumed to be
+LCNames. In "Term" fields (`648`, `650`, `651`), it is assumed to be LCSH.
+
+This is not reliable.  Many "Geographical Terms" are in fact Names of Places.
+
+I believe that this could be resolved by examining the first character of the identifier (n for LCNames, s for LCSH).
+
+### Compound Concepts
+
+Compound concepts in MARC data from Sierra bear an identifier referring to the whole concept, but the transformer
+also breaks them up into the individual constituent concepts.  However, the identifiers for these constituent concepts 
+are only "label-derived".  LoC and MeSH entries for compound concepts also contain the identifiers for the constituents,
+so a new stage could populate these properly.
+
+### Validation
+
+Currently, there is no validation as to whether the given identifier actually exists within the given scheme.
+Even if it does, there is no validation as to whether there is any correspondence between the name of the concept
+and the identifier.
+
+As a result, there are some errors that can currently only be detected by eye.
+
+* Some MeSH identifiers have been found wrongly assigned to the LCSH scheme.  
+* Two LCNames identifiers (n50034502 and n2001003970) do not exist
+* Some identifiers in the LCSH scheme have been parsed/written incorrectly, so they are not identifiers at all.
+* The wrong field was being extracted from some TEI records to find the LCSH id.
+
+In addition, there may be some that are very difficult to detect by eye.  A real identifier that refers to something different
+to the label of the field would only be detected when someone follows a link and finds themselves on a wholly unexpected Concept page.
+`n2001003970`, mentioned above, does not exist but `nr2001003970` does.  
+However, that identifer refers to the "Partnership for Governance Reform in Indonesia", whereas
+the associated label in the offending Work is "Pseudo Aristotele., De coloribus., Latin".
```

Comment:
Interesting - would question whether we should be doing this as part of the pipeline, perhaps more appropriate to script some identifier validation so we can pass affected records on to collections info

Reply:
I think there are two modes for this 

- find all the broken ones and get someone to fix them
- don't let any new broken ones get through the pipeline

If we can add identifier validation even earlier, so that whoever is typing it in gets a warning immediately, then that would be nice.

Reply:
We've talked in the past about the pipeline having a way to signal errors in the data, of which this would definitely be an example.